tests/integration_determinism.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+import numpy as np
+import torch
+from torch.utils.data import DataLoader, Dataset
+
+from monai.data.synthetic import create_test_image_2d
+from monai.losses.dice import DiceLoss
+from monai.networks.nets.unet import UNet
+
+
+def run_test(batch_size=64, train_steps=100, device=torch.device("cuda:0")):
+
+    class _TestBatch(Dataset):
+
+        def __getitem__(self, _unused_id):
+            im, seg = create_test_image_2d(128, 128, noise_max=1, num_objs=4, num_seg_classes=1)
+            return im[None], seg[None].astype(np.float32)
+
+        def __len__(self):
+            return train_steps
+
+    net = UNet(
```

Comment:
I remember the densenet pipeline also works well in the demo, how could we make it reproducible? but maybe that's a network testing task

Reply:
all reasonable experiments should be another integration task.